pkg/authz/check/service.go

```diff
@@ -100,6 +99,59 @@ func (svc CheckService) getMatchingSubjects(ctx context.Context, objectType stri
 	return warrantSpecs, nil
 }
 
+func (svc CheckService) getMatchingSubjectsWithSubjectType(ctx context.Context, objectType string, objectId string, relation string, subjectType string, wntCtx wntContext.ContextSetSpec) ([]warrant.WarrantSpec, error) {
+	log.Debug().Msgf("Getting matching subjects for %s:%s#%s@%s:___%s", objectType, objectId, relation, subjectType, wntCtx)
+
+	warrantSpecs := make([]warrant.WarrantSpec, 0)
+	objectTypeSpec, err := svc.ObjectTypeSvc.GetByTypeId(ctx, objectType)
+	if err != nil {
+		return warrantSpecs, err
+	}
+
+	if _, ok := objectTypeSpec.Relations[relation]; !ok {
+		return warrantSpecs, nil
+	}
+
+	warrants, err := svc.WarrantRepository.GetAllMatchingObjectAndRelationWithSubjectType(
+		ctx,
+		objectType,
+		objectId,
+		relation,
+		subjectType,
+		wntCtx.ToHash(),
+	)
+	if err != nil {
+		log.Err(err).Msg("Error fetching warrants for object")
+		return warrantSpecs, err
+	}
+
+	for _, warrant := range warrants {
+		warrantSpecs = append(warrantSpecs, *warrant.ToWarrantSpec())
+	}
+
+	if err != nil {
+		return warrantSpecs, err
+	}
+
+	warrants, err = svc.WarrantRepository.GetAllMatchingWildcard(
```

Comment:
In both this method and in `getMatchingSubjects`, do we still need to call `GetAllMatchingWildcard` now that we're performing the wildcard match in `GetAllMatchingObjectAndRelationWithSubjectType` and `GetAllMatchingObjectAndRelation` respectively?

Reply:
No, I think we can remove this completely now that we've updated the queries in `GetAllMatchingObjectAndRelation` and `GetAllMatchingObjectAndRelationWithSubjectType`.